zeeguu_api/api/user_languages.py

```diff
@@ -0,0 +1,155 @@
+        user_language.declared_level = language_level
+    session.add(user_language)
+    session.commit()
+
+    return "OK"
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{DELETE_USER_LANGUAGE}/<language_id>", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def delete_user_language(language_id):
+    """
+    A user can delete a language with a given ID.
+    With this endpoint the full user language with all the data is deleted.
+    At the moment not used but might be useful.
+
+    :return: "OK" in case of success
+    """
+
+    try:
+        to_delete = UserLanguage.with_language_id(language_id, flask.g.user)
+        session.delete(to_delete)
+        session.commit()
+    except Exception as e:
+        return "OOPS. SEARCH AIN'T THERE IT SEEMS (" + str(e) + ")"
+
+    return "OK"
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{USER_LANGUAGES}", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def get_user_languages():
+    """
+    A user might be subscribed to multiple languages at once.
+    This endpoint returns them as a list.
+
+    :return: a json list with languages for which the user is registered;
+     every language in this list is a dictionary with the following info:
+                id = unique id of the language;
+                language = <unicode string>
+    """
+    all_user_languages = []
+    user_languages = UserLanguage.all_for_user(flask.g.user)
+    for lan in user_languages:
+        all_user_languages.append(lan.as_dictionary())
+    return json_result(all_user_languages)
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{READING_LANGUAGES}", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def get_reading_languages():
+    """
+    A user might be subscribed to multiple languages at once.
+    This endpoint returns them as a list.
+
+    :return: a json list with languages for which the user is registered;
+     every language in this list is a dictionary with the following info:
+                id = unique id of the language;
+                language = <unicode string>
+    """
+    all_user_languages = []
+    reading_languages = UserLanguage.all_reading_for_user(flask.g.user)
+    for lan in reading_languages:
+        all_user_languages.append(lan.as_dictionary())
+    return json_result(all_user_languages)
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{INTERESTING_LANGUAGES}", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def get_interesting_languages():
```

Comment:
interesting_languages_for_reading

because we will also have language info about exercises,
right?